sklearn/utils/validation.py

```diff
@@ -728,6 +852,15 @@ def check_array(
 
         .. versionadded:: 1.1.0
 
+    asarray : callable, default=None
+        A callable to use instead of `np.asarray` when converting the input
+        array. Useful when the input array is not a Numpy array or when the
+        converted array should be a ndarray from a differnt library. The callable
+        should have the same signature as `np.asarray` and in addition support
+        they `copy` keyword argument.
```

Comment:
Can the function signature be written out here? `asarray(a, dtype=None, order=None, copy=False)`

([np.asarray](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html) has a `like` kwarg which is not required)

Reply:
Yes, I'm still thinking about what the best way is to describe how this `asarray`-like function should behave. Both in terms of its signature and what it can and can not return (your later comment).